CMakeLists.txt

```diff
@@ -320,10 +320,14 @@ if (WITH_SYSTEM_PROVIDED_3PARTY)
   set(GFLAGS_USE_TARGET_NAMESPACE ON)
   find_package(gflags REQUIRED)
 
+  find_package(PkgConfig)
+  pkg_check_modules(jansson REQUIRED IMPORTED_TARGET jansson)
+  add_library(jansson::jansson ALIAS PkgConfig::jansson)
```

Comment:
How did it work on other systems without explicit PkgConfig specification? Will it continue to work? Why it doesn't find the library without explicit PkgConfig?

CC @Ferenc-

Reply:
Normally, jansson is vendored as a 3party and configured with CMake:
https://github.com/organicmaps/organicmaps/blob/master/3party/CMakeLists.txt#L35

This will continue to work. A jansson installation of a distribution normally contains (just) a pkgconfig file (it's pretty common). See for example the debian package: https://packages.debian.org/bookworm/amd64/libjansson-dev/filelist

> Why it doesn't find the library without explicit PkgConfig?

Appearently, CMake's `find_package` does not search for pkgconfig files. It needs the `FindPkgConfig` module for that: https://cmake.org/cmake/help/latest/module/FindPkgConfig.html

Reply:
Does it mean that on Debian/Ubuntu the current master works with PkgConfig for utf8 even without explicitly writing it in cmake? If yes, why? Could it be that cmake detects some pkgconfig vars/files/configs on Debian/Ubuntu, but doesn't do it on Gentoo for some reason? Maybe some env vars are missing?

cmake has `--trace` parameter that may be helpful to discover issues with packages.

Reply:
Do you mean the test pipeline here? Does it build with or without `WITH_SYSTEM_PROVIDED_3PARTY`?